client/feature_cache.go

```diff
@@ -45,3 +49,26 @@ func (fc *FeatureCache) Get(name string) *Feature {
 
 	return feature
 }
+
+func (fc *FeatureCache) GetAll() map[string]*Feature {
+	fc.lock.RLock()
+	defer fc.lock.RUnlock()
+
+	// deep clone features map
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+	dec := gob.NewDecoder(&buf)
+
+	err := enc.Encode(fc.features)
+	if err != nil {
+		return nil
```

Comment:
This error is not possible, because you know features will always be able to encode properly. This is fine, but I personally prefer to crash the application with a `panic` that to fail silently and have unexpected behaviour in another part of the system.